src/main/java/Models/Appointment/Appointment.java

```diff
@@ -0,0 +1,286 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package Models.Appointment;
+
+import java.sql.SQLException;
+import java.sql.Time;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ *
+ * @author maxwell
+ */
+public class Appointment {
+
+    public ArrayList<Event> sortAppointmentsByDate(String appointmentDate, ArrayList<Event> event) {
+        ArrayList<Event> e = new ArrayList<>();
+
+        event.stream().peek((Event event1) -> System.out.println(event1.getDate())).filter((event1) -> (event1.getDate().equals(appointmentDate))).forEachOrdered(e::add);
+        return e;
+    }
+
+    public ArrayList<String> getGeneratedSlots(int apptmLength, String doctorName, String date) {
+        ArrayList<Event> currEvents;
+        ArrayList<String> intervals = new ArrayList<>();
+        try {
+            String[] hours = new String[2];
+            boolean addSlot;
+            SimpleDateFormat timeFormat = new SimpleDateFormat("HH:mm");
+
+            currEvents = AppointmentDAO.retrieveDoctorsAppointment(doctorName, true);
+            currEvents = sortAppointmentsByDate(date, currEvents);
+
+            Calendar cal = Calendar.getInstance();
+            Calendar cal2 = Calendar.getInstance();
+            cal.setTime(timeFormat.parse("9:00"));
+            cal2.setTime(timeFormat.parse("18:00"));
+            Time u = new java.sql.Time(cal.getTimeInMillis());
```

Comment:
This likely isn't a requirement but might be something to be aware of.

At my previous job, we ran into issues with this method (Or similar) when working with multiple time zones. For example, if a patient were to book a holiday online while abroad. This function (Or similar) would return the local time and make the whole journey go a bit spoopy.

Maybe we could talk about this when going for those 80+ marks in the final sprint.

Reply:
Dealing with timezones is always a nightmare lol